lib/crc32.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Microsemi Switchtec(tm) PCIe Management Library
+ * Copyright (c) 2019, Microsemi Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#ifndef CRC32_H
+#define CRC32_H
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#define CRC32_WIDTH             (8 * sizeof(uint32_t))    /* in bits  */
+
+/*
+* Standard CRC-32 Polynomial of:
+*      x^32+x^26+x^23+x^22+x^16+x^12+x^11+x^10+x^8+x^7+x^5+x^4+x^2+x^1+1
+*/
+#define CRC32_POLYNOMIAL        (0x04C11DB7)
+#define CRC32_INIT_REMAIDER     (0xFFFFFFFF)
+#define CRC32_FINAL_XOR_VALUE   (0xFFFFFFFF)
+
+uint32_t pmc_crc32(const uint8_t *msg_ptr, uint32_t byte_cnt,
+		   uint32_t oldchksum, int init, int last);

```

Comment:
As this is the only thing for this header, I'd suggest we add it to an existing header. switchtec_priv.h or switchtec.h maybe.

Also, please add the CRC in a separate commit with it's own description.

Reply:
I can put this in common.h but since we already have a 'crc8.h' which is the same situation, I would like to maintain consistency for now.

All other suggestions are adopted and code updated. @lsgunth please have a look.